crates/rune/src/indexing/index.rs

```diff
@@ -712,7 +740,7 @@ impl Index for ast::ItemFn {
             };
 
             idx.query.insert_meta(span, meta)?;
-        } else if is_toplevel && item.visibility.is_public() {
+        } else if is_toplevel && item.visibility.is_public() || is_test {
```

Comment:
this isn't enough to make #[test] in mods auto-discovered, we need to be even more eager about them.